### PR TITLE
Add tenant to notification payload

### PR DESCRIPTION
**Pull request by ivantenevvasilev**
**Description**

Changes proposed in this pull request:
- add `TenantWithLabels` to `ApplicationWithLabels`,  `ApplicationTemplateWithLabels`, `RuntimeWithLabels` and `RuntimeContextWithLabels`.
- split webhook input builder into multiple files
- add unit tests
- add new e2e test

**Pull Request status**

- [x] Implementation
- [x] Unit tests
- [x] Integration tests
- [x] `chart/compass/values.yaml` is updated 
- [x] Mocks are regenerated, using the automated script

### CODE REVIEW

File: components/director/pkg/graphql/formation_constraint_validation.go [hunk 1]
```diff
@@ -85,20 +85,32 @@ func emptyGenerateFormationAssignmentNotificationOperationDetails() *formationco
 		ApplicationTemplate: &webhook.ApplicationTemplateWithLabels{
 			ApplicationTemplate: &model.ApplicationTemplate{},
 			Labels:              map[string]string{},
+			Tenant: &webhook.TenantWithLabels{
```
[the remark this answers is not in the document]
Comment (ivantenevvasilev) [reply components/director/pkg/graphql/formation_constraint_validation.go:88]: I forgot to adapt it here, initially it was missing in `webhook_validation.go`, but it was causing problems in the `inputTemplate`, so I added it there. I just forgot to add it in this validation as well.

File: components/director/internal/domain/webhook/datainputbuilder/builder.go [hunk 2]
```diff
@@ -97,13 +116,20 @@ func (b *WebhookDataInputBuilder) PrepareApplicationAndAppTemplateWithLabels(ctx
 		if err != nil {
 			return nil, nil, errors.Wrapf(err, "while getting application template with ID: %q", *application.ApplicationTemplateID)
 		}
-		applicationTemplateLabels, err := b.getLabelsForObject(ctx, tenant, appTemplate.ID, model.AppTemplateLabelableObject)
+		applicationTemplateLabels, err := b.labelInputBuilder.GetLabelsForObject(ctx, tenant, appTemplate.ID, model.AppTemplateLabelableObject)
+		if err != nil {
+			return nil, nil, errors.Wrapf(err, "while building labels for application template with ID %q", appTemplate.ID)
+		}
+
+		tenantsWithLabelsForApplicationTemplate, err := b.tenantInputBuilder.GetTenantForApplicationTemplate(ctx, tenant, applicationTemplateLabels)
```
[the remark this answers is not in the document]
Comment (ivantenevvasilev) [reply components/director/internal/domain/webhook/datainputbuilder/builder.go:124]: Yes, it should be as you suggested, my bad. I will adapt it.

**Comment by nyordanoff** [comment]
reset-vm

**Review by nyordanoff — DISMISSED** [review]

**Review by alextargov — APPROVED** [review]

**Comment by nyordanoff** [comment]
reset-vm

**Comment by nyordanoff** [comment]
reset-vm

**Review by nyordanoff — APPROVED** [review]

**Review by alextargov — APPROVED** [review]

**Pull request merged**